packages/webview_flutter/webview_flutter/lib/webview_flutter.dart

```diff
@@ -634,19 +635,29 @@ class WebViewController {
 
   /// Loads the specified URL.
   ///
-  /// If `headers` is not null and the URL is an HTTP URL, the key value paris in `headers` will
-  /// be added as key value pairs of HTTP headers for the request.
+  /// If [WebViewRequest] object is not null and the URL is a HTTP URL
+  /// then the following rules apply:
+  ///
+  /// - [WebViewRequest.method] must be one of the supported HTTP methods
+  /// in the [WebViewLoadMethod].
+  ///
+  /// - If [WebViewRequest.headers] is not null, the key value pairs in
+  /// [WebViewRequest.headers] will be added as key value pairs of HTTP headers
+  /// for the request.
+  ///
+  /// - If [WebViewRequest.body] is not null, it will be added as HTTP body
+  /// for the request.
   ///
   /// `url` must not be null.
   ///
   /// Throws an ArgumentError if `url` is not a valid URL string.
   Future<void> loadUrl(
     String url, {
-    Map<String, String>? headers,
+    WebViewRequest? request,
```

Comment:
All the same comments apply here (I meant to make them on the public interface not the platform interface, but they are applicable in both places.)